packages/lab/src/__tests__/__e2e__/tabs-next/TabsNext.cy.tsx

```diff
@@ -0,0 +1,417 @@
+          cy.get(".saltTabstripNext").invoke("css", "width", "320px");
+          cy.get(".saltTabstripNext-inner")
+            .findAllByRole("tab")
+            .should("have.length", 5)
+            .eq(4)
+            .should("not.be.visible");
+          cy.findByRole("combobox").should("exist").click();
+          cy.findByRole("listbox")
+            .findAllByRole("option")
+            .should("have.length", 1);
+        }
+      );
+    });
+  });
+});
+
+describe("Tab selection, Given a Tabstrip", () => {
+  describe("WHEN initial size is sufficient to display all contents", () => {
+    describe("WHEN it initially renders", () => {
+      describe("WHEN the selected Tab has not been specified", () => {
+        it("THEN the first tab will be selected", () => {
+          cy.mount(<SimpleTabstrip width={400} />);
+          cy.findAllByRole("tab").eq(0).should("have.ariaSelected");
+        });
+      });
+    });
+  });
+
+  describe("WHEN initial size is NOT sufficient to display all contents", () => {
+    describe("WHEN the selected Tab is in the overflow menu", () => {
+      it("THEN the active tab will be moved from the overflow menu to the end of visible tabs", () => {
+        cy.mount(<SimpleTabstrip width={220} />);
+        cy.get(".saltTabstripNext-inner [role='tab']:first-child").should(
+          "have.ariaSelected"
+        );
+
+        cy.findByRole("combobox").click();
+        cy.findByRole("listbox");
+        cy.focused().realPress("ArrowDown").realPress("Enter");
+        cy.findAllByRole("tab").eq(1).should("have.ariaSelected");
+      });
+    });
+    describe("WHEN the selected Tab is moved to visible", () => {
+      describe("WHEN the component is resized", () => {
+        it("THEN the active tab will be moved from the overflow menu to the end of visible tabs", () => {
+          cy.mount(<SimpleTabstrip width={320} />);
+          cy.get(".saltTabstripNext-inner [role='tab']:first-child").should(
+            "have.ariaSelected"
+          );
+          cy.findByRole("combobox").click();
+          cy.findByRole("listbox");
+          cy.focused().realPress("ArrowDown").realPress("Enter");
+          cy.findAllByRole("tab").eq(2).should("have.ariaSelected");
+          cy.get(".saltTabstripNext").invoke("css", "width", "220px");
+          cy.findAllByRole("tab").eq(1).should("have.ariaSelected");
+        });
+      });
+    });
+  });
+});
+
+describe("Navigation, Given a Tabstrip", () => {
+  describe("WHEN initial size is sufficient to display all contents", () => {
+    describe("WHEN the tabstrip is first rendered", () => {
+      describe("WHEN the tabstrip receives keyboard focus", () => {
+        it("THEN focus will be transfered to the first tab", () => {
+          cy.mount(
+            <StackLayout>
+              <button data-testid="tabstop-1" />
+              <SimpleTabstrip width={400} />
+              <button data-testid="tabstop-2" />
```

Comment:
are the additional buttons necessary?

Reply:
i think we can do without the buttons in some of these tests. In this one you're only testing tabbing in, so there's no need for them. We don't do this in the other components, we Tab straight in. I don't mind the technique, I'm only mentioning this so we can have consistency through all our tests